zircon.core/common/src/main/kotlin/org/hexworks/zircon/internal/component/impl/DefaultSlider.kt

```diff
@@ -0,0 +1,254 @@
+        }
+    }
+
+    private fun setValueToClosestOfStep(step: Int) {
+        var actualStep = step
+        if (step < 0) {
+            actualStep = 0
+        }
+        val calculatedValue = actualStep * valuePerStep
+        currentValue = calculatedValue.roundToInt()
+    }
+
+    abstract fun getMousePosition(event: MouseEvent): Int
+
+    override fun acceptsFocus() = isDisabled.not()
+
+    override fun focusGiven() = whenEnabled {
+        LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was given focus.")
+        componentStyleSet.applyFocusedStyle()
+        render()
+    }
+
+    override fun focusTaken() = whenEnabled {
+        LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) lost focus.")
+        componentStyleSet.reset()
+        render()
+    }
+
+    override fun mouseEntered(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse entered.")
+            componentStyleSet.applyMouseOverStyle()
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mouseExited(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse exited.")
+            componentStyleSet.reset()
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mousePressed(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse pressed.")
+            componentStyleSet.applyActiveStyle()
+            setValueToClosestOfStep(getMousePosition(event))
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mouseReleased(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse released.")
+            componentStyleSet.applyMouseOverStyle()
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun mouseDragged(event: MouseEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            LOGGER.debug("Gutter (id=${id.abbreviate()}, disabled=$isDisabled) was mouse pressed.")
+            componentStyleSet.applyActiveStyle()
+            setValueToClosestOfStep(getMousePosition(event))
+            render()
+            Processed
+        } else Pass
+    }
+
+    override fun keyPressed(event: KeyboardEvent, phase: UIEventPhase) = whenEnabledRespondWith {
+        if (phase == UIEventPhase.TARGET) {
+            when (event.code) {
+                KeyCode.RIGHT -> {
```

Comment:
Will this be used for both the vertical and the horizontal implementation? Having `RIGHT` increment makes sense for a horizontal slider, but not so much for a vertical one.

Reply:
You're right, should I lift that method to each of the children classes ?